### PR TITLE
Use AsSpan/AsMemory (start, length) 

### PR DESCRIPTION
**Pull request by xtqqczze**
* Replace `AsSpan().Slice(start, length)` with `AsSpan(start, length)`
* Replace `AsMemory().Slice(start, length)` with `AsMemory(start, length)`

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Net.Security/tests/StressTests/SslStress/StressOperations.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ public DataSegment Deserialize(ReadOnlySequence<byte> buffer)
 
             ReadOnlySequence<byte> lengthBytes = buffer.Slice(0, pos.Value);
             int numSize = s_encoding.GetChars(lengthBytes.ToArray(), _charBuffer);
-            int length = int.Parse(_charBuffer.AsSpan().Slice(0, numSize));
+            int length = int.Parse(_charBuffer.AsAsSpan(0, numSize));
```
Comment (stephentoub) [inline src/libraries/System.Net.Security/tests/StressTests/SslStress/StressOperations.cs:150]:
```suggestion
            int length = int.Parse(_charBuffer.AsSpan(0, numSize));
```

File: src/libraries/System.Net.Security/tests/StressTests/SslStress/StressOperations.cs [hunk 2]
```diff
@@ -159,7 +159,7 @@ public DataSegment Deserialize(ReadOnlySequence<byte> buffer)
 
             ReadOnlySequence<byte> checksumBytes = buffer.Slice(0, pos.Value);
             numSize = s_encoding.GetChars(checksumBytes.ToArray(), _charBuffer);
-            ulong checksum = ulong.Parse(_charBuffer.AsSpan().Slice(0, numSize));
+            ulong checksum = ulong.Parse(_charBuffer.AsAsSpan(0, numSize));
```
Comment (stephentoub) [inline src/libraries/System.Net.Security/tests/StressTests/SslStress/StressOperations.cs:162]:
```suggestion
            ulong checksum = ulong.Parse(_charBuffer.AsSpan(0, numSize));
```

**Review by GrabYourPitchforks — APPROVED** [review]
Normally I'd suggest __not__ making changes to unit test files, as we could inadvertently bypass legitimate tests. In the case of _AsSpan.cs_, there are unit tests that validate that `AsSpan().Slice(...)` and `AsSpan(...)` have the same result, and doing a search-and-replace of these patterns will end up essentially removing the value that such tests would've given us.

That said, it looks like in the case of _GuidArrayAsSpanWithStartAndLength_ the test actually did intend to test `AsSpan(...)` instead of `AsSpan().Slice(...)`, if we're following the same pattern used by the other tests whose names end in the suffix _\*AsSpanWithStartAndLength_.

But in the future please try to limit these changes to non-test code. Thanks!

**Comment by xtqqczze** [comment]
> In the case of _AsSpan.cs_, there are unit tests that validate that `AsSpan().Slice(...)` and `AsSpan(...)` have the same result, and doing a search-and-replace of these patterns will end up essentially removing the value that such tests would've given us.

@GrabYourPitchforks These tests were not part of the merged changes?

**Comment by xtqqczze** [comment]
> But in the future please try to limit these changes to non-test code. Thanks!

Yeah I did think twice about making these changes to tests, but in the case of _GuidArrayAsSpanWithStartAndLength_ and others, they made sense.

**Pull request merged**